Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer+Playback.swift

```diff
@@ -87,8 +87,10 @@ extension AudioPlayer {
         let time = time.clamped(to: 0...duration)
 
         if status == .playing {
+            isSeeking = true
```

Comment:
This should work for now, although I'm wondering if it would be possible in the future to have this be part of the player status enums. Also, I wasn't able to find a data race you mentioned, but I'm curious now where that happens. Does Xcode say which line it happens on?

Reply:
So the race, when I attempted to add the `if isSeeking { return }` logic back inside `internalCompletionHandler` is like:

- we set `isSeeking`
- we dispatch the call to `internalCompletionHandler` onto the main thread
- we unset `isSeeking`
- now `internalCompletionHandler` fires on the main thread, but we're not actively seeking.

Reply:
Ah I see. Yeah I found it best to avoid running certain tasks in the completion handler because it actually runs on a background serial queue. Synchronizing resources can be tricky. The way you did it in the PR should be fine now. I'm going to rework some more code and clean up that completion handler so we can be careful about async tasks.

Let me know if you see anything else which needs polishing, and thanks for catching this one 🙌🏻!